micro_dl/preprocessing/resize_images.py

```diff
@@ -107,3 +113,66 @@ def resize_frames(self):
             os.path.join(self.resize_dir, "frames_meta.csv"),
             sep=',',
         )
+
+    def resize_volumes(self, num_slices_subvolume=-1):
+        """Down or up sample volumes
+
+        :param int num_slices_subvolume: num of 2D slices to include in each
+         volume. if -1, include all slices
+        """
+
+        # assuming slice_ids will be continuous
+        num_total_slices = len(self.slice_ids)
+        if not isinstance(self.scale_factor, float):
+            sc_str = '-'.join(self.scale_factor.astype('str'))
+        else:
+            sc_str = self.scale_factor
+
+        mp_args = []
+        resized_metadata_list = []
+        if num_slices_subvolume == -1:
+            num_slices_subvolume = len(self.slice_ids)
+        num_blocks = np.ceil(
+            num_total_slices / num_slices_subvolume
+        ).astype('int')
+        for time_idx in self.time_ids:
+            for pos_idx in self.pos_ids:
+                for channel_idx in self.channel_ids:
+                    for slice_idx in range(num_blocks):
+                        start_idx = slice_idx * num_slices_subvolume
```

Comment:
When there's no overlap in z between blocks, will there be discontinuities between the upsampled blocks?

Reply:
Yes, there will be a discontinuity of one slice! Thanks for catching this, the index here is also wrong (assumes slice_idx starting from 0)....